### PR TITLE
Deprecate support for go 1.14 and go 1.15

### PR DESCRIPTION
**Pull request by gaby**
* Update all go.mod files to use `go 1.19`
* Removed `go 1.14 and go 1.15` from all the test cases
* Added `go 1.18` to all the drivers tests.

Note:
* I had to remove `go 1.16` from the test matrix in `test.yml` (local storage). The reason is because badger no longer supports 1.16 as seen in #608 

Fixes #608 #548

### CODE REVIEW

**Comment by efectn** [comment]
I think we can split .github/workflows/test.yml tests

**Comment by gaby** [comment]
> I think we can split .github/workflows/test.yml tests

Sounds good! I'd add that to this PR

File: .github/workflows/test-sqlite3.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+on:
+    push:
+        branches:
+            - master
+            - main
+    pull_request:
+name: "Tests Local Storage"
```
Comment (ReneWerner87) [inline .github/workflows/test-sqlite3.yml:7]: don't forget to update the badges links in the readme in the root folder

**Review by efectn — APPROVED** [review]

File: .github/workflows/test-badger.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+on:
+    push:
+        branches:
+            - master
+            - main
+    pull_request:
+name: "Tests Badger"
```
Comment (ReneWerner87) [inline .github/workflows/test-badger.yml:7]: don't forget to update the badges links in the readme in the root folder

**Pull request merged**